index.js

```diff
@@ -87,6 +107,22 @@ MarketplaceClient.prototype.validatePackage = function(packagePath) {
         }, function(error, response, body) {
           if (error) {
             reject(error);
+          } else if (response.statusCode == 202) {
+            // Checks the validation's status every five seconds, if it's not yet processed
+            var poller = function() {
+              setTimeout(function() {
+                self.getValidation(body.id).then(function(validation) {
+                  if (validation.processed) {
+                    resolve(validation);
+                  } else {
+                    poller();
+                  }
+                }, function() {
```

Comment:
If there's a problem with the request (i.e. no network connection, the server sends an error, etc.) this will still keep polling until the user cancels it though, right? Does Marketplace's API often cause the Promise to reject? If not maybe a better error handler could be used here.